File: Perl/examples/PaymentAPI/getSettlements.pl
#!/usr/bin/perl
use strict;
use warnings;
use JSON::PP;
use Data::Dumper;
use lib '../..';
require "PaymentAPI.pl";
require "LoadEnv.pl";
LoadEnv('../../.env');

my $test = 1;
my $debug = 0;

# Credentials for Auth
my $id = $ENV{"EID"};
my $key = $ENV{"SECRET"};

my $api = PaymentAPI->new($id, $key, $test, $debug);
my $values = {
    "fromDate" => "2024-03-31",
};
print(Dumper($api->call("getSettlements", $values)));

1;

File: Deno/examples/PaymentAPI/getSettlements.ts
import {QvicklyPaymentAPI, env} from "../../PaymentAPI.ts";

const paymentAPI = new QvicklyPaymentAPI(env["EID"], env["SECRET"]);
const settlements = await paymentAPI.call("getSettlements", {
  fromDate: "2024-03-01",
});
console.log(settlements);
